In [1]:
import pandas as pd

In [2]:
ipl=pd.read_csv("ipl_2023_dataset.csv",usecols=["Player Name", "Type", "Price Cr", "Team"])

In [3]:
ipl.head()

Player Name           Type  Price Cr                 Team
0       Ben Stokes     All-Rounder     16.25  Chennai Super Kings
1  Ravindra Jadeja     All-Rounder     16.00  Chennai Super Kings
2    Deepak Chahar          Bowler     14.00  Chennai Super Kings
3         MS Dhoni   Wicket-Keeper     12.00  Chennai Super Kings
4        Moeen Ali     All-Rounder      8.00  Chennai Super Kings

In [4]:
ipl.describe()

Price Cr
count  243.000000
mean     3.746914
std      4.584679
min      0.200000
25%      0.225000
50%      1.500000
75%      6.375000
max     18.500000

In [5]:
ipl.loc[(ipl["Price Cr"]==18.5)]

Player Name         Type  Price Cr          Team
146  Sam Curran   All-Rounder      18.5  Punjab Kings

Sam Curran was sold at the highest bid of RS18.500 Cr

In [6]:
Lowest_price_of_players=ipl.loc[(ipl["Price Cr"]==0.20)]

In [7]:
print(Lowest_price_of_players)

              Player Name           Type  Price Cr                 Team
17   Matheesha Pathirana          Bowler       0.2  Chennai Super Kings
18      Mukesh Choudhary          Bowler       0.2  Chennai Super Kings
19         Shaik Rasheed          Batter       0.2  Chennai Super Kings
20       Simarjeet Singh          Bowler       0.2  Chennai Super Kings
21   Subhranshu Senapati          Batter       0.2  Chennai Super Kings
..                    ...            ...       ...                  ...
217         Kuldip Yadav          Bowler       0.2     Rajasthan Royals
239         Sanvir Singh     All-Rounder       0.2  Sunrisers Hyderabad
240         Samarth Vyas     All-Rounder       0.2  Sunrisers Hyderabad
241   Nitish Kumar Reddy   Wicket-Keeper       0.2  Sunrisers Hyderabad
242     Anmolpreet Singh          Batter       0.2  Sunrisers Hyderabad

[61 rows x 4 columns]


Above given are the players who are been sold at the lowest price of Rs 20 Lakh 

In [8]:
ipl.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 568 entries, 0 to 567
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Player Name  568 non-null    object 
 1   Type         568 non-null    object 
 2   Price Cr     243 non-null    float64
 3   Team         568 non-null    object 
dtypes: float64(1), object(3)
memory usage: 17.9+ KB


In [9]:
ipl.tail()

Player Name         Type  Price Cr    Team
563        Dhruv Patel  All-Rounder       NaN  Unsold
564    Jack Prestwidge  All-Rounder       NaN  Unsold
565     Aditya Sarvate  All-Rounder       NaN  Unsold
566      Sagar Solanki  All-Rounder       NaN  Unsold
567  Prenelan Subrayen  All-Rounder       NaN  Unsold

In [10]:
ipl.dtypes

Player Name     object
Type            object
Price Cr       float64
Team            object
dtype: object

In [11]:
ipl.isnull().sum()

Player Name      0
Type             0
Price Cr       325
Team             0
dtype: int64

In [12]:
ipl.loc[(ipl["Team"]=="Unsold")].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 325 entries, 243 to 567
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Player Name  325 non-null    object 
 1   Type         325 non-null    object 
 2   Price Cr     0 non-null      float64
 3   Team         325 non-null    object 
dtypes: float64(1), object(3)
memory usage: 12.7+ KB


SO the numbers of players that are sold in the ipl auction 2022 is 243

In [13]:
568-325 

243

568 is the total number of players

Total number of players sold in the auction is 243

325 players unsold

In [14]:
players_sold=ipl.dropna()

In [15]:
print(players_sold)

             Player Name           Type  Price Cr                 Team
0            Ben Stokes     All-Rounder     16.25  Chennai Super Kings
1       Ravindra Jadeja     All-Rounder     16.00  Chennai Super Kings
2         Deepak Chahar          Bowler     14.00  Chennai Super Kings
3              MS Dhoni   Wicket-Keeper     12.00  Chennai Super Kings
4             Moeen Ali     All-Rounder      8.00  Chennai Super Kings
..                   ...            ...       ...                  ...
238       Upendra Yadav   Wicket-Keeper      0.25  Sunrisers Hyderabad
239        Sanvir Singh     All-Rounder      0.20  Sunrisers Hyderabad
240        Samarth Vyas     All-Rounder      0.20  Sunrisers Hyderabad
241  Nitish Kumar Reddy   Wicket-Keeper      0.20  Sunrisers Hyderabad
242    Anmolpreet Singh          Batter      0.20  Sunrisers Hyderabad

[243 rows x 4 columns]


In [16]:
players_sold["Team"].unique()

array(['Chennai Super Kings', 'Delhi Capitals', 'Gujarat Titans',
       'Lucknow Super Giants', 'Mumbai Indians', 'Punjab Kings',
       'Royal Challengers Bangalore', 'Rajasthan Royals',
       'Sunrisers Hyderabad'], dtype=object)

In [40]:
CSK=ipl.loc[(ipl["Team"]=="Chennai Super Kings")]
CSK_players_name_value=CSK.iloc[:,[2]]
CSK_players_name_value_array=CSK_players_name_value.to_numpy()
CSK_players_name_value_array


array([[16.25],
       [16.  ],
       [14.  ],
       [12.  ],
       [ 8.  ],
       [ 6.75],
       [ 6.  ],
       [ 4.  ],
       [ 1.9 ],
       [ 1.5 ],
       [ 1.2 ],
       [ 1.  ],
       [ 1.  ],
       [ 0.7 ],
       [ 0.6 ],
       [ 0.5 ],
       [ 0.5 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ]])

In [18]:
DC=ipl.loc[(ipl["Team"]=="Delhi Capitals")]
DC_players_name_value=DC.iloc[:,[2]]
DC_players_name_value_array=DC_players_name_value.to_numpy()
DC_players_name_value_array

In [19]:
GT=ipl.loc[(ipl["Team"]=="Gujarat Titans")]
GT_players_name_value=CSK.iloc[:,[2]]
CSK_players_name_value_array=CSK_players_name_value.to_numpy()
CSK_players_name_value_array

In [20]:
LSG=ipl.loc[(ipl["Team"]=="Lucknow Super Giants")]

In [21]:
MI=ipl.loc[(ipl["Team"]=="Mumbai Indians")]

In [22]:
PK=ipl.loc[(ipl["Team"]=="Punjab Kings")]

In [23]:
RCB=ipl.loc[(ipl["Team"]=="Royal Challengers Bangalore")]

In [24]:
RR=ipl.loc[(ipl["Team"]=="Rajasthan Royals")]

In [25]:
SH=ipl.loc[(ipl["Team"]=="Sunrisers Hyderabad")]

In [26]:
import matplotlib.pyplot as plt
%matplotlib inline

In [27]:
players_sold.head()

Player Name           Type  Price Cr                 Team
0       Ben Stokes     All-Rounder     16.25  Chennai Super Kings
1  Ravindra Jadeja     All-Rounder     16.00  Chennai Super Kings
2    Deepak Chahar          Bowler     14.00  Chennai Super Kings
3         MS Dhoni   Wicket-Keeper     12.00  Chennai Super Kings
4        Moeen Ali     All-Rounder      8.00  Chennai Super Kings

In [28]:
CSK.head()

Player Name           Type  Price Cr                 Team
0       Ben Stokes     All-Rounder     16.25  Chennai Super Kings
1  Ravindra Jadeja     All-Rounder     16.00  Chennai Super Kings
2    Deepak Chahar          Bowler     14.00  Chennai Super Kings
3         MS Dhoni   Wicket-Keeper     12.00  Chennai Super Kings
4        Moeen Ali     All-Rounder      8.00  Chennai Super Kings

In [29]:
CSK_players_name=CSK.iloc[:,[0]]
CSK_players_name_array=CSK_players_name.to_numpy()
CSK_players_name

Player Name
0               Ben Stokes 
1          Ravindra Jadeja 
2            Deepak Chahar 
3                 MS Dhoni 
4                Moeen Ali 
5            Ambati Rayudu 
6          Ruturaj Gaikwad 
7              Shivam Dube 
8         Mitchell Santner 
9   Rajvardhan Hangargekar 
10        Prashant Solanki 
11           Kyle Jamieson 
12            Devon Conway 
13      Maheesh Theekshana 
14          Nishant Sindhu 
15          Ajinkya Rahane 
16        Dwaine Pretorius 
17     Matheesha Pathirana 
18        Mukesh Choudhary 
19           Shaik Rasheed 
20         Simarjeet Singh 
21     Subhranshu Senapati 
22           Bhagath Varma 
23             Ajay Mandal 
24        Tushar Deshpande

array([[16.25],
       [16.  ],
       [14.  ],
       [12.  ],
       [ 8.  ],
       [ 6.75],
       [ 6.  ],
       [ 4.  ],
       [ 1.9 ],
       [ 1.5 ],
       [ 1.2 ],
       [ 1.  ],
       [ 1.  ],
       [ 0.7 ],
       [ 0.6 ],
       [ 0.5 ],
       [ 0.5 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ]])

In [35]:
DC.head()

Player Name           Type  Price Cr            Team
25    Rishabh Pant   Wicket-Keeper      16.0  Delhi Capitals
26      Axar Patel     All-Rounder      12.0  Delhi Capitals
27    Prithvi Shaw          Batter       8.0  Delhi Capitals
28  Mitchell Marsh     All-Rounder       6.5  Delhi Capitals
29   Anrich Nortje          Bowler       6.5  Delhi Capitals

In [36]:
DC_players_price=DC.iloc[:,[2]]

In [37]:
DC_players_price.to_numpy()

array([[16.  ],
       [12.  ],
       [ 8.  ],
       [ 6.5 ],
       [ 6.5 ],
       [ 6.25],
       [ 5.5 ],
       [ 5.25],
       [ 4.6 ],
       [ 4.2 ],
       [ 2.8 ],
       [ 2.4 ],
       [ 2.  ],
       [ 2.  ],
       [ 2.  ],
       [ 1.1 ],
       [ 0.65],
       [ 0.5 ],
       [ 0.5 ],
       [ 0.5 ],
       [ 0.5 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ]])

In [39]:
GT_players_name_value=GT.iloc[:,[2]]
GT_players_name_value_array=GT_players_name_value.to_numpy()
GT_players_name_value_array

array([[15.  ],
       [15.  ],
       [ 9.  ],
       [ 8.  ],
       [ 6.25],
       [ 6.  ],
       [ 4.4 ],
       [ 3.2 ],
       [ 3.  ],
       [ 3.  ],
       [ 2.6 ],
       [ 2.4 ],
       [ 2.4 ],
       [ 2.  ],
       [ 1.9 ],
       [ 1.7 ],
       [ 1.4 ],
       [ 1.2 ],
       [ 0.5 ],
       [ 0.5 ],
       [ 0.3 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ],
       [16.  ],
       [12.25],
       [12.  ],
       [10.75],
       [10.  ],
       [ 8.  ],
       [ 8.  ],
       [ 6.  ],
       [ 2.  ],
       [ 1.5 ],
       [ 1.5 ],
       [ 1.  ],
       [ 0.9 ],
       [ 0.6 ],
       [ 0.55],
       [ 0.5 ],
       [ 0.5 ],
       [ 0.5 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ],
       [ 0.2 ]])

In [ ]:
GT